protocols/kad/src/behaviour.rs

```diff
@@ -2054,8 +2109,17 @@ pub struct PeerRecord {
 /// See [`NetworkBehaviour::poll`].
 #[derive(Debug)]
 pub enum KademliaEvent {
-    /// A query has produced a result.
-    QueryResult {
+    /// An inbound request has been received and handled.
+    //
+    // Note on the difference between 'request' and 'query': A request is a
+    // single request-response style exchange with a single remote peer. A query
+    // is made of multiple requests across multiple remote peers.
+    InboundRequest {
```

Comment:
The name of this variant is not very "eventy" because it doesn't contain a verb and hence doesn't really express, what happened.

What about `InboundRequestServed`? That would also include the information you've packed into the docs.